### PR TITLE
Enable building with Address Sanitizer

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Enable building with Address Sanitizer.

Build and run unit_test and ebpf_client test.

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
Note: Need to fix the issues that Address Sanitizer is flagged.

**Comment by Alan-Jowett** [comment]
Example failure:
https://github.com/microsoft/ebpf-for-windows/runs/5620221460?check_suite_focus=true

File: .github/workflows/build Address Sanitizer.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      uses: microsoft/setup-msbuild@v1
+
+    - name: Install ProcDump64
+      working-directory: ${{env.GITHUB_WORKSPACE}}
+      run: |
+        curl -fsSL -o Procdump.zip https://download.sysinternals.com/files/Procdump.zip
+        7z x Procdump.zip -y -o"C:/Program Files/ProcDump"
+        echo "C:\Program Files\ProcDump" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+        mkdir c:/dumps/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}
+        echo "test" > c:/dumps/${{env.BUILD_PLATFORM}}/${{env.BUILD_CONFIGURATION}}/test.txt
```
Comment (dthaler) [inline .github/workflows/build Address Sanitizer.yml:45]: What's the purpose of line 45?  If it's needed then can we add a comment somewhere about why it's needed?

**Review by dthaler — APPROVED** [review]

File: .github/workflows/build Address Sanitizer.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+      BUILD_PLATFORM: x64
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+
+    - name: Add MSBuild to PATH
+      uses: microsoft/setup-msbuild@v1
```
Comment (dthaler) [inline .github/workflows/build Address Sanitizer.yml:36]:
```suggestion
      uses: microsoft/setup-msbuild@becb80cf9a036187bb1e07e74eb64e25850d757a
```

File: .github/workflows/build Address Sanitizer.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        cmake -G "Visual Studio 16 2019" -S external\catch2 -B external\catch2\build -DBUILD_TESTING=OFF
+
+    - name: Build
+      working-directory: ${{env.GITHUB_WORKSPACE}}
+      # Add additional options to the MSBuild command line here (like platform or verbosity level).
+      # See https://docs.microsoft.com/visualstudio/msbuild/msbuild-command-line-reference
+      run: msbuild /m /p:Configuration=${{env.BUILD_CONFIGURATION}} /p:Platform=${{env.BUILD_PLATFORM}} ${{env.SOLUTION_FILE_PATH}} /p:AddressSanitizer='True' /t:tests\unit_tests:Rebuild /t:tests\ebpf_client:Rebuild
+
+    - name: Upload Build Output
+      uses: actions/upload-artifact@v2.2.4
```
Comment (dthaler) [inline .github/workflows/build Address Sanitizer.yml:79]:
```suggestion
      uses: actions/upload-artifact@27121b0bdffd731efa15d66772be8dc71245d074
```

**Review by dthaler — APPROVED** [review]

**Pull request merged**